# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1251373594 -7200
#      Thu Aug 27 13:46:34 2009 +0200
# Node ID 8bf27324a9d0b8eabb101a5bef6cc2c60c24ef12
# Parent  42a18d5aff7fcc2eb2a2a14ced6530203caf1f98
fix current class method determination

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-27  Jaroslav Hajek  <highegg@gmail.com>
+
+	* src/ov-class.cc (get_current_method_class): Simplify.
+
 2009-08-27  Jaroslav Hajek  <highegg@gmail.com>
 
 	* OPERATORS/op-cs-cs.cc: Simplify comparison operators.
 	* OPERATORS/op-fcs-fcs.cc: Ditto.
 
 2009-08-25  David Grundberg  <individ@acc.umu.se>
 
 	* DLD-FUNCTIONS/__magick_read__.cc [HAVE_MAGICK]: Include
diff --git a/src/ov-class.cc b/src/ov-class.cc
--- a/src/ov-class.cc
+++ b/src/ov-class.cc
@@ -118,21 +118,17 @@ octave_class::unique_clone (void)
 
 static std::string
 get_current_method_class (void)
 {
   std::string retval;
 
   octave_function *fcn = octave_call_stack::current ();
 
-  std::string my_dir = fcn->dir_name ();
-
-  std::string method_class = file_ops::tail (my_dir);
-
-  retval = method_class.substr (1);
+  retval = fcn->dispatch_class ();
 
   return retval;
 }
 
 static void
 gripe_invalid_index (void)
 {
   error ("invalid index for class");
